app/src/main/java/org/fossasia/openevent/general/social/SocialLink.kt

```diff
@@ -0,0 +1,24 @@
+package org.fossasia.openevent.general.social
+
+import android.arch.persistence.room.*
+import android.arch.persistence.room.ForeignKey.CASCADE
+import com.fasterxml.jackson.databind.PropertyNamingStrategy
+import com.fasterxml.jackson.databind.annotation.JsonNaming
+import com.github.jasminb.jsonapi.IntegerIdHandler
+import com.github.jasminb.jsonapi.annotations.Id
+import com.github.jasminb.jsonapi.annotations.Relationship
+import com.github.jasminb.jsonapi.annotations.Type
+import org.fossasia.openevent.general.event.Event
+
+@Type("social-link")
+@JsonNaming(PropertyNamingStrategy.KebabCaseStrategy::class)
+@Entity(foreignKeys = [(ForeignKey(entity = Event::class, parentColumns = ["id"], childColumns = ["event"], onDelete = CASCADE))])
+data class SocialLink(
+        @Id(IntegerIdHandler::class)
+        @PrimaryKey
+        val id: Int,
+        val link: String,
+        val name: String,
+        @Relationship("event")
+        var event: Event?
```

Comment:
Why var and nullable?

Reply:
I was facing an error which gave that parameter specified as non-null is null

Reply:
This means that response is not receiving event as relationship. Fix that instead of removing this

Reply:
that is because you havent added include event statement inside the API

Reply:
Its still returning that error, event is always null

Reply:
Check the logs for why is that

Reply:
Yeah I did check, it like the jackson mapping is not done properly

Reply:
` java.lang.RuntimeException: com.fasterxml.jackson.databind.JsonMappingException: Instantiation of [simple type, class org.fossasia.openevent.general.social.SocialLink] value failed (java.lang.IllegalArgumentException): Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkParameterIsNotNull, parameter event
06-12 18:08:18.409 32538-32538/org.fossasia.openevent.general E/SocialLinksViewModel$loadSocialLinks: `

Reply:
Create a new type called EventId which only contains id attribute and use it instead of whole Event